### PR TITLE
Wrote script to restore hidden repo if parity is lost

### PR DESCRIPTION
**Pull request by breillyrh**

### CODE REVIEW

File: bin/dock-pulp-recreate-hidden.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            die('  Missing %s repository, please see dock-pulp-bootstrap' % \
+                dockpulp.HIDDEN)
+        else:
+            raise
+    log.info('  %s is present' % dockpulp.HIDDEN)
+    # start copy of all images to hidden repo
+    for repo in repos:
+        if repo['id'] == dockpulp.HIDDEN:
+            continue
+        for img in repo['images'].keys():
```
Comment (twaugh) [inline bin/dock-pulp-recreate-hidden.py]: Do we need to do this for manifests as well? Does copying a manifest also copy the blobs it references?

Comment (breillyrh) [inline bin/dock-pulp-recreate-hidden.py]: Yes, I'll be adding manifest support today. And thankfully, pulp is smart enough to copy blobs when a manifest is copied.

File: bin/dock-pulp-recreate-hidden.py [hunk 2]
```diff
@@ -85,6 +86,15 @@ def recreate(dpo, jfile, test=False):
             else:
                 p.copy(dockpulp.HIDDEN, img, source=repo['id'])
 
+        # set default in case env has no v2 content
+        repo.setdefault('manifests', {})
+
+        for manifest in repo['manifests'].keys():
+            if test:
+                log.info('  Would have copied %s from %s to hidden repo', manifest, repo['id'])
```
Comment (twaugh) [inline bin/dock-pulp-recreate-hidden.py]: Should this say 'Would have copied manifest %s ...' to distinguish it?

Comment (breillyrh) [inline bin/dock-pulp-recreate-hidden.py]: Yes, fair point.

**Comment by twaugh** [comment]
LGTM

**Pull request merged**